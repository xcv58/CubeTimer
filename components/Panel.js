import React from 'react'
import { inject, observer } from 'mobx-react'
import Button from '@material-ui/core/Button'

@inject('store')
@inject('recordsStore')
@observer
export default class Panel extends React.Component {
  clear = () => {
    const { recordsStore, store } = this.props
    recordsStore.clear()
    store.clear()
  }

  render () {
    const { recordsStore: { records }, store: { running, lapse } } = this.props
    if (records.length < 1 && (running || lapse === 0)) {
      return null
    }
    return (
      <div className='timebar' style={{
        display: 'flex',
        justifyContent: 'center'
      }}>
        <Button onClick={this.clear}
          disabled={running}
          style={{ width: '100%' }}>
          Clear
        </Button>
      </div>
    )
  }
}
