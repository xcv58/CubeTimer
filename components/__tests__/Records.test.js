/* global it, expect, describe */
import React from 'react'
import Records, { Record } from '../Records'
import Panel from '../Panel'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Record', () => {
  it('render correct text', () => {
    const el = shallow(<Record {...{ lapse: 42, min: 1, index: 1 }} />)
    expect(el.text().includes('00:00:04')).toBe(true)
    expect(el.text().endsWith('00:00:04')).toBe(true)
    expect(el.text().startsWith(1)).toBe(true)
    expect(true)
  })

  it('render correct style', () => {
    let el = shallow(<Record {...{ lapse: 42, min: 1, index: 1 }} />)
    expect(el.find('div').props().style).toBe(undefined)
    let tree = renderer.create(
      <Record {...{ lapse: 42, min: 1, index: 1 }} />
    ).toJSON()
    expect(tree).toMatchSnapshot()

    el = shallow(<Record {...{ lapse: 42, min: 42, index: 1 }} />)
    expect(el.find('div').is({ style: { color: 'black' } })).toBe(true)
    tree = renderer.create(
      <Record {...{ lapse: 42, min: 42, index: 1 }} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Records', () => {
  it('render Panel', () => {
    const recordsStore = { records: [] }
    const props = { recordsStore }
    const el = shallow(<Records {...props} />).dive()
    expect(el.find(Panel).exists()).toBe(true)
  })

  it('render list of Record', () => {
    const recordsStore = {
      min: 42,
      records: [
        { lapse: 42, timestamp: 1 },
        { lapse: 153, timestamp: 2 }
      ]
    }
    const props = { recordsStore }
    const el = shallow(<Records {...props} />).dive()
    expect(el.find(Record).length).toBe(recordsStore.records.length)
  })
})
