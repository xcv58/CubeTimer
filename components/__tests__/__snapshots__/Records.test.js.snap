// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record render correct style 1`] = `
<div
  className="timebar"
>
  <span
    style={
      Object {
        "float": "left",
        "width": 0,
      }
    }
  >
    1
  </span>
  00:00:04
</div>
`;

exports[`Record render correct style 2`] = `
<div
  className="timebar"
  style={
    Object {
      "color": "black",
    }
  }
>
  <span
    style={
      Object {
        "float": "left",
        "width": 0,
      }
    }
  >
    1
  </span>
  00:00:04
</div>
`;
